agent/consul/discoverychain/gateway.go

```diff
@@ -90,8 +94,10 @@ func (l *GatewayChainSynthesizer) AddHTTPRoute(route structs.HTTPRouteConfigEntr
 			}
 		}
 
-		l.matchesByHostname[host] = matches
+		currentMatches[host] = matches
 	}
+	//TODO def don't think this is needed just testing for now, remove if not needed
+	return currentMatches
```

Comment:
It looks like this is needed given `l.matchesByHostname` appears to be passed in by value? Probably cleaner to keep it this way rather than mutating an array passed into this function by reference.

Reply:
Removing this TODO